src/core/lightning/index.ts

```diff
@@ -102,31 +83,18 @@ export const LightningMixin = (superclass) =>
         throw new LndOfflineError("no active lnd to create an invoice")
       }
 
-      try {
-        input = {
-          lnd,
-          tokens: value,
-          expires_at,
-        }
+      const lndService = MakeLndService(lnd)
+      const result: ResultAsync<RegisteredInvoice, LightningServiceError> =
+        lndService.registerInvoice({
+          description: typeof memo == "string" ? memo : "",
```

Comment:
why do we need to check type here? the type must be validated in the param (IAddInvoiceRequest)